examples/heatTransfer/read/heatRead_adios2.cpp

```diff
@@ -60,12 +60,14 @@ int main(int argc, char *argv[])
         // ISO-POSIX file is the default transport
         // Passing parameters to the transport
         bpReaderSettings.AddTransport("File", "verbose=4");
+        bpReaderSettings.SetParameters("OpenAsFile");
     }
 
     auto bpReader = ad.Open(inputfile, "r", mpiReaderComm, bpReaderSettings);
 
     if (bpReader == nullptr)
-        throw std::ios_base::failure("ERROR: failed to open ADIOS bpReader\n");
+        throw std::ios_base::failure("ERROR: failed to open " +
+                                     std::string(inputfile) + "\n");
 
     unsigned int gndx;
```

Comment:
prefer 
unsigned int gndx = vgnx->m_Data[0]; as in line 79
See: https://github.com/ornladios/ADIOS2/wiki/Variables-Scope,-Functions,-and-Namespaces
2. Variable declaration